test/integration/cli-args/stanc.t

```diff
@@ -22,6 +22,7 @@ Show help
     --version                       Display stanc version number
     --name                          Take a string to set the model name (default = "$model_filename_model")
     --O                             Allow the compiler to apply all optimizations to the Stan code.
+    -fno-soa                        Turn off the Struct of Arrays memory optimization.
```

Comment:
We never really had a good plan for these stanc3 flags, so we are a bit all over the place with them. 

I am not sure I have a better idea, but the `-fno-X` seems like a C++ compiler flag or is this style also common in other areas?

@WardBrian already brought up the idea of enabling individual optimization flags. Do we maybe want to have the optimization levels but then a flag to disable individual optimizatons?

Reply:
We can also merge this without discussing this, there is plenty of time before the release.

Reply:
Yes so I think what we want to do is 

1. Merge #1079 which has some of the code in the PR that fixes something in the monotone framework 
2. Merge #549 so we have an O1
3. Merge this and turn it on at O1

Personally I'd prefer if we merged this immediately after the latest release. While I'm confident in it, I'd rather have a release cycle on dev just to make sure try to find any weird quirks

Reply:
Cool with that yeah, although if its hidden behind O1, its less of a problem.